org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/ClasspathChange.java

```diff
@@ -416,6 +418,7 @@ public int generateDelta(JavaElementDelta delta, boolean addClasspathChange) {
 						}
 					}
 				}
+				addAttributeDeltas(delta, this.oldResolvedClasspath[i], newResolvedClasspath[index]);
```

Comment:
So we see attribute change only for existing classpath entry. The old (removed) classpath entry that could contain attributes will not have them in the delta?
This seem to be OK & repeats what was done for source attachment changes.